### ISSUE TITLE
ZeroMQPubServerChannel memory leak when TCP transport minion connects to 0mq master

### ISSUE THREAD

**Issue by ahammond** [saltstack/salt#36612]
### Description of Issue/Question

Restarted salt-master yesterday afternoon. This morning:

```
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root     17529  0.6 26.2 8971700 8648380 ?     Sl   Sep26   6:33 /usr/bin/python /usr/bin/salt-master ZeroMQPubServerChannel
```

Note 26.2% memory usage on a 32G Amazon instance is about 8GB. Captured a coredump in the hopes this will be useful. Restart salt-master, wait a few hours for it to stabilize:

```
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root     21123  0.3  2.5 1217244 833700 ?      Sl   17:29   0:12 /usr/bin/python /usr/bin/salt-master ZeroMQPubServerChannel
```

Note 2.5% memory usage.

### Setup
I'm not sure what specific stuff in our setup is relevant. We just switched back from TCP to 0mq transports. I'm pretty sure we have a couple of machines that didn't get re-configured (the reason we're switching is that we needed more reliable behavior).

### Steps to Reproduce Issue
We have about 500 minions connected with 0mq. I'm guessing we still have a few that are trying to connect with TCP, which I'll be hunting down today. Hopefully eliminating them will eliminate the issue, in which case I'll update the issue description and title.

### Versions Report
```
ahammond@salt:~$ sudo salt-call --local --versions-report
Salt Version:
           Salt: 2016.3.3

Dependency Versions:
           cffi: 1.5.2
       cherrypy: 2.3.0
       dateutil: 2.4.2
          gitdb: 0.6.4
      gitpython: 2.0.6
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.3.7
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Jul  1 2016, 15:12:24)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: 0.2.4
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.4.0-36-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

@white-hat @onorua @jgartrel

**Comment by Ch3LL** [comment]
@ahammond does the fix in #36618 fix this issue?

**Comment by ahammond** [comment]
Yup, my co-worker developed it specifically to fix this. If it's in 2016.3
then we can close this.

On Sep 28, 2016 9:11 AM, "Megan Wilhite" <notifications@github.com> wrote:

@ahammond <https://github.com/ahammond> does the fix in #36618
<https://github.com/saltstack/salt/pull/36618> fix this issue?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/saltstack/salt/issues/36612#issuecomment-250215568>,
or mute
the thread

**Comment by rallytime** [comment]
The back-port for the fix to go into 2016.3 is in #36636, but it's still running tests.

**Comment by ahammond** [comment]
Awesome, I'll close this in favor of #36636
